tests/unit/swagger/definitionsGeneration/definitions.spec.ts

```diff
@@ -318,6 +318,29 @@ describe('Definition generation', () => {
                     expect(propertySchema).to.not.haveOwnProperty('additionalProperties', `for property ${propertyName}`);
                     expect(propertySchema['x-nullable']).to.eq(true, `for property ${propertyName}[x-nullable]`);
                 },
+                objLiteral: (propertyName, propertySchema) => {
+                    expect(propertySchema).to.deep.include({
+                      properties: {
+                        name: {
+                          type: 'string',
+                        },
+                        nested: {
+                          properties: {
+                            allOptional: {
```

Comment:
I was unaware that swagger let you specify a type within a type. I thought it only allowed $refs. I’m assuming you’ve checked that the output is valid Schema within the “pet shop” Swagger editor?

If it really is that that simple then this PR is going to make a lot of people happy! :)

Reply:
I checked [swagger editor](https://editor.swagger.io) and [swagger parser](https://apidevtools.org/swagger-parser/online/).
Just like you, I expected to move this to the OAPI3 generator but tried with the OAPI2 generator mostly to find hints what the errors would be. Turns out no errors :smile: 

For convenience, here is a small-ish YAML:
```yaml
basePath: /v1
consumes:
  - application/json
definitions:
  EnumIndexValue:
    description: EnumIndexValue.
    enum:
      - '0'
      - '1'
    type: string
  TestModel:
    description: This is a description of a model
    properties:
      objLiteral:
        properties:
          name:
            type: string
          nested:
            properties:
              bool:
                type: boolean
              optional:
                type: number
                format: double
              allOptional:
                properties:
                  one:
                    $ref: '#/definitions/EnumIndexValue'
                  two:
                    type: string
                type: object
            required:
              - allOptional
              - bool
            type: object
        required:
          - name
        type: object
    required:
      - objLiteral
info:
  title: Test API
  version: 1.0.0
  description: Description of a test API
  license:
    name: MIT
paths:
  /GetTest:
    get:
      operationId: GetModel
      produces:
        - application/json
      responses:
        '200':
          description: Returns TestModel
          schema:
            $ref: '#/definitions/TestModel'
          examples:
            application/json:
              objLiteral:
                name: a string
                nested:
                  bool: true
      description: |-
        This is a description of the getModel method
        this is some more text on another line
      security: []
      parameters: []
produces:
  - application/json
swagger: '2.0'
securityDefinitions:
  basic:
    type: basic
host: 'localhost:3000'
```